File: packages/sdk-node/src/retry.ts
const BASE_DELAY_MS = 500;
const JITTER_MAX_MS = 100;

export interface RetryConfig {
  maxRetries: number;
  timeout: number;
}

export function isRetryable(status: number): boolean {
  return status === 429 || status >= 500;
}

export function getRetryDelay(attempt: number, retryAfterHeader: string | null): number {
  if (retryAfterHeader) {
    const seconds = parseInt(retryAfterHeader, 10);
    if (!isNaN(seconds)) return seconds * 1000;
  }
  const exponential = BASE_DELAY_MS * Math.pow(2, attempt);
  const jitter = Math.random() * JITTER_MAX_MS;
  return exponential + jitter;
}

export function sleep(ms: number): Promise<void> {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

File: packages/sdk-python/src/alrt_sdk/retry.py
"""Retry logic with exponential backoff."""
import random
import time
import asyncio

BASE_DELAY = 0.5  # 500ms
JITTER_MAX = 0.1  # 100ms


def is_retryable(status: int) -> bool:
    return status == 429 or status >= 500


def get_retry_delay(attempt: int, retry_after: str | None) -> float:
    if retry_after:
        try:
            return float(retry_after)
        except ValueError:
            pass
    exponential = BASE_DELAY * (2 ** attempt)
    jitter = random.random() * JITTER_MAX
    return exponential + jitter


def sync_sleep(seconds: float) -> None:
    time.sleep(seconds)


async def async_sleep(seconds: float) -> None:
    await asyncio.sleep(seconds)
